src/components/FormikField.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import BEMHelper from 'react-bem-helper';
+import { Field, connect } from 'formik';
+
+export const classes = new BEMHelper({
+  name: 'field',
+  prefix: 'c-',
+});
+
+const FormikFieldLabel = ({ label, noBorder, name }) => {
+  if (!label) {
+    return null;
+  }
+  if (!noBorder) {
+    return <label htmlFor={name}>{label}</label>;
+  }
+  return (
+    <label className="u-hidden" htmlFor={name}>
+      {label}
+    </label>
+  );
+};
+
+FormikFieldLabel.propTypes = {
+  noBorder: PropTypes.bool,
+  name: PropTypes.string.isRequired,
+  label: PropTypes.string,
+};
+
+export const FormikFieldError = ({ children }) => (
+  <span {...classes('help', 'error')}>{children}</span>
+);
+
+const FormikField = ({
+  children,
+  className,
+  label,
+  name,
+  noBorder,
+  title,
+  right,
+  formik,
+  showError,
+  ...rest
+}) => {
+  return (
+    <div {...classes('', { 'no-border': noBorder, right, title }, className)}>
+      <FormikFieldLabel label={label} name={name} noBorder={noBorder} />
+      <Field name={name} {...rest}>
+        {children || null}
+      </Field>
+      {showError && label && formik.errors[name] && (
+        <FormikFieldError>{formik.errors[name](label)}</FormikFieldError>
+      )}
+    </div>
+  );
+};
+
+FormikField.propTypes = {
+  noBorder: PropTypes.bool,
+  right: PropTypes.bool,
+  title: PropTypes.bool,
+  name: PropTypes.string.isRequired,
+  label: PropTypes.string,
+  showError: PropTypes.bool,
+  formik: PropTypes.shape({
+    form: PropTypes.shape({
+      errors: PropTypes.shape({}),
+      touched: PropTypes.shape({}),
+    }),
+  }),
+};
+
+FormikField.defaultProps = {
+  noBorder: false,
+  showError: true,
+};
+
+export default connect(FormikField);
```

Comment:
Ganske sikker på at man bør enten bruke render props eller hoc, ikke begge deler. Og alt du trenger å sende ned er jo bare errors? Viss du hadde klart å fjerne behovet for errors som en funksjon av label så kunne du også brukt <ErrorMessage fra Formik, og ikke trengt å sende ned errors heller

Reply:
Nja, er jo det connect er til for. At man kan hooke på context'en fra formik i hvilken som helst komponenent. Tenker du på withFormik kanskje?

https://jaredpalmer.com/formik/docs/api/connect